tests/apps/courses/test_models_course_run.py

```diff
@@ -22,6 +22,16 @@ def setUp(self):
         super().setUp()
         self.now = timezone.now()
 
+    def test_models_course_run_state_coming_soon(self):
+        """
+        A course run that has no value passed should return a state with 
+        "coming soon" as text.
+        """
+        course_run = CourseRunFactory(start=None)
+        self.assertEqual(
+            course_run.state, CourseState(6, None, "coming soon", None)
```

Comment:
In this case we should also make sure that it is ranked last by the method course.get_course_runs. I'm afraid they might come first at the moment.
Or should we exclude them from the search engine index until they are properly defined?
Another soliution could be to introduce a new state for unscheduled courses so that it's easier to rank them last.
WDYT?

Reply:
Introducing a new state would be ideal. The new indexing is done on courses instead of course-run so we'll need a way to represent it anyway. What does @mbenadda think about this ?

Reply:
According to info on the state property in course:
src/richie/apps/courses/models/course.py line 210

The CourseState 6 is used when there is no courserun and is the lowest priority. This means that we can use this state as well as long as there is no date entered...

Reply:
Good idea. 
BTW the implications of such change are deep because they impact the search engine. As we are in the middle of a big refactoring of the search engine, I would like to postpone this work...